consensus/hotstuff/signature/weighted_signature_aggregator_test.go

```diff
@@ -238,4 +242,174 @@ func TestWeightedSignatureAggregator(t *testing.T) {
+			_, err = aggregator.TrustedAdd(ids[1].NodeID, sigs[1])
+			require.NoError(t, err)
+
+			// Aggregation should error with sentinel InvalidSignatureIncludedError
+			// aggregated public key is identity
+			signers, agg, err := aggregator.Aggregate()
+			assert.True(t, model.IsInvalidSignatureIncludedError(err))
+			assert.Nil(t, agg)
+			assert.Nil(t, signers)
+		})
+	})
+
+	//  3. Signatures serialization is valid but some signatures are invalid w.r.t their respective public keys.
+	//      3.a. aggregated public key is not identity
+	//      3.b. aggregated public key is identity
+	t.Run("correct serialization and invalid signature", func(t *testing.T) {
+		ids, pks, sigs, msg, _, tag := createAggregationData(t, 2)
+
+		t.Run("with non-identity aggregated public key", func(t *testing.T) {
+			aggregator, err := NewWeightedSignatureAggregator(ids, pks, msg, tag)
+			require.NoError(t, err)
+
+			// add a valid signature
+			err = aggregator.Verify(ids[0].NodeID, sigs[0])
+			require.NoError(t, err)
+			_, err = aggregator.TrustedAdd(ids[0].NodeID, sigs[0])
+			require.NoError(t, err)
+
+			// add invalid signature for signer with index 1
+			// sanity check: Verify should reject it
+			err = aggregator.Verify(ids[1].NodeID, sigs[0])
+			assert.ErrorIs(t, err, model.ErrInvalidSignature)
+			_, err = aggregator.TrustedAdd(ids[1].NodeID, sigs[0])
+			require.NoError(t, err)
+
+			// Aggregation should error with sentinel InvalidSignatureIncludedError
+			// aggregated public key is not identity (equal to pk[0] + pk[1])
+			signers, agg, err := aggregator.Aggregate()
+			assert.Error(t, err)
+			assert.True(t, model.IsInvalidSignatureIncludedError(err))
+			assert.Nil(t, agg)
+			assert.Nil(t, signers)
+		})
+
+		t.Run("with identity aggregated public key", func(t *testing.T) {
+			// assign  pk1 to -pk0 so that the aggregated public key is identity
+			// this is a shortcut since PoPs are not checked in this test
+			pkBytes := pks[0].Encode()
+			negatePoint(pkBytes)
+			var err error
+			pks[1], err = crypto.DecodePublicKey(crypto.BLSBLS12381, pkBytes)
+			require.NoError(t, err)
+
+			aggregator, err := NewWeightedSignatureAggregator(ids, pks, msg, tag)
+			require.NoError(t, err)
+
+			// add a valid signature for index 0
+			err = aggregator.Verify(ids[0].NodeID, sigs[0])
+			require.NoError(t, err)
+			_, err = aggregator.TrustedAdd(ids[0].NodeID, sigs[0])
+			require.NoError(t, err)
+
+			// add an invalid signature for signer with index 1
+			// sanity check: Verify should reject it
+			err = aggregator.Verify(ids[1].NodeID, sigs[0])
+			assert.ErrorIs(t, err, model.ErrInvalidSignature)
+			_, err = aggregator.TrustedAdd(ids[1].NodeID, sigs[0])
+			require.NoError(t, err)
+
+			// Aggregation should error with sentinel InvalidAggregatedKeyError
+			// aggregated public key is identity
+			signers, agg, err := aggregator.Aggregate()
+			assert.Error(t, err)
+			assert.True(t, model.IsInvalidAggregatedKeyError(err))
```

Comment:
same here, the test could pass if either `InvalidSignatureIncludedError` or  `InvalidAggregatedKeyError` is returned